website/server/models/challenge.js

```diff
@@ -149,7 +154,7 @@ schema.methods.syncToUser = async function syncChallengeToUser (user) {
     }
   });
 
-  toSave.push(user.save());
+  toSave.push(User.collection.update({_id: user._id}, user.toObject()));
```

Comment:
I think the best thing to do is to use `concat` instead of `push` on the challenges array as outlined here https://github.com/HabitRPG/habitrpg/pull/7787#issuecomment-232972394 at the end.

I would like to point out, though, that mongoose should protect against this type of issues and I'm not sure why we have this issues.

So while the change is okay I'd add a comment above it to explain while we're doing it (that mongoose's protection against concurrent updates of arrays using the __v field isn't working as expected)